### PR TITLE
Added isindex function.

### LINKED ISSUE #30: isindex function
It would be useful to have a function that checks if an object is an index type. It should have a flag to include or not include NDIndex types.

### PR DESCRIPTION
**Pull request by melissawm**
Please check if this is the right format for the "exclude" flag.

Closes #30

### CODE REVIEW

**Comment by melissawm** [comment]
The tests are failing on master and are unrelated.

File: ndindex/__init__.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 __all__ = []
 
-from .ndindex import ndindex
+from .ndindex import ndindex, isindex
 
 __all__ += ['ndindex']
```
Comment (asmeurer) [inline ndindex/__init__.py:5]: You need to add it to `__all__`

File: ndindex/tests/test_ndindex.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import inspect
 
-from hypothesis import given, example
+from hypothesis import given, example, note
```
Comment (asmeurer) [inline ndindex/tests/test_ndindex.py:3]: This import is unused.

**Comment by asmeurer** [comment]
It should also include the base index types. For integer indices the most correct thing is to check if `operator.index` raises an exception or not.

**Comment by asmeurer** [comment]
By "base" I mean built-in, like tuple, slice, int, etc., as well as valid index numpy arrays (is it possible numpy already has such a function?).

**Comment by melissawm** [comment]
Sorry it took me so long, let me know if this is enough.

**Comment by scopatz** [comment]
This looks good to me. CC @asmeurer

**Comment by scopatz** [comment]
It looks like there are some conflicts here now

File: ndindex/ndindex.py [hunk 3]
```diff
@@ -30,21 +31,77 @@ def ndindex(obj):
...
+    False
+    """
+    from . import ellipsis, Integer
+
+    try:
+        Integer(obj)
+        if exclude and ((int in exclude) or (Integer in exclude)):
+            return False
+        else:
+            return True
```
Comment (scopatz) [inline ndindex/ndindex.py:70]: It seems like this if-statement doesn't need to be in the try-block

**Comment by melissawm** [comment]
Closing this as I assume it's not relevant anymore, cheers!

**Pull request closed without merging**